irc: add support for IRCv3.2 chghost

sim642 opened the pull request.

maxteufel commented:
This should contain an option to make host changes visible in the channel buffer.

sim642 commented:
@maxteufel Sensible idea, made it happen. If one doesn't want to see chghost messages they can always easily be filtered with the `irc_chghost` tag, just like joins/quits.

Techman- commented:
So, what's the status on this?

miniCruzer commented:
Also wondering the same. (:

sim642 commented:
It's just waiting to be merged along with bunch of other IRCv3.2 PRs. I just rebased it to make it compile again after API changes and added `chghost` capability to be properly mentioned too.

Techman- commented:
@sim642 Well, I'm hoping it gets added soon, because I think it's been a good while :)

miniCruzer commented:
Applied this patch to my WeeChat, and soon after squashed an ircd side bug in this capability. Thanks @sim642!

https://github.com/charybdis-ircd/charybdis/commit/412263854f1ec526a79fd4015dd5f833d72e0c93
